addon/components/frost-autocomplete.js

```diff
@@ -0,0 +1,125 @@
+import Ember from 'ember'
+import computed, {readOnly} from 'ember-computed-decorators'
+import {PropTypes} from 'ember-prop-types'
+import layout from '../templates/components/frost-autocomplete'
+import {keyCodes} from '../utils'
+import FrostSelect from './frost-select'
+
+const {$, get, isEmpty, run} = Ember
+const {DOWN_ARROW, UP_ARROW} = keyCodes
+
+export default FrostSelect.extend({
+  layout,
+  classNames: 'frost-select',
+
+  propTypes: {
+    // options
+    isLoading: PropTypes.bool,
+
+    // state
+    focusedIndex: PropTypes.number,
+    _userInput: PropTypes.bool,
+    filterType: PropTypes.oneOf(['startsWith', 'contains']),
+    _autofocus: PropTypes.bool,
+    disableSpaceToggle: PropTypes.bool
+  },
+
+  getDefaultProps () {
+    return {
+      focusedIndex: 0,
+      _userInput: false,
+      filterType: 'startsWith',
+      isLoading: false,
+      disabled: false,
+      _autofocus: false,
+      disableSpaceToggle: true
+    }
+  },
+
+  /* eslint-disable complexity */
```

Comment:
We should not be employing `eslint-disable complexity` in new code.  It should only be used to temporarily address existing code to work around additions or refactors but in new code it should just be written correctly.